### DISCUSSION TITLE
Docker/Podman and persistent users/shares

### DISCUSSION THREAD

**Discussion by rotten777** [filebrowser/filebrowser#2160] [General]
So I'm  using podman on an Ubuntu based host and can get everything up, add users, upload files, publish shares, etc.  but on reboots I lose the users passwords and shares. 

The relevant parts of the system service I have setup for filebrowser is:

```
ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --rm --sdnotify=conmon --replace -t -d --name filebrowser -v /srv/filebrowser:/srv -v /srv/filebrowser.db:/database/filebrowser.db -v /srv/settings.json:/config/settings.json -u 0:0 -p 127.0.0.1:8080:80 docker.io/filebrowser/filebrowser
```

**/srv/filebrowser.db**: is an empty file with world writable permissions.
**/srv/filebrowser**: directory with all of the uploaded files from all users; this persists no problem
**/srv/settings.json**: world readable file, contents below

```
{
  "port": 8080,
  "baseURL": "https://files.mydomain.tld",
  "address": "127.0.0.1",
  "log": "stdout",
  "database": "/database/filebrowser.db",
  "root": "/srv"
}

```
I'm struggling to see where the issue is. The user data (username, password hash, etc.)  is stored in the filebrowser.db file, correct? Is there somewhere that logs storage errors where I can see the user data stored?

I should add the only thing my container logs give me:
```
2022/10/12 12:30:07 Using config file: /.filebrowser.json
2022/10/12 12:30:07 Listening on [::]:80
2022/10/12 12:30:28 /api/renew: 401 50.194.214.99 <nil>
```

**Comment by ramiresviana** [comment]
As seen in the logs, the default config file located at ` /.filebrowser.json` is being used. Since there's no volume to persist the default database location at `/database.db`, it is lost when restarting the container. Using the `config` option allows telling Filebrowser which config file should be used, check https://filebrowser.org/cli/filebrowser.
